include/tvm/relax/expr.h

```diff
@@ -37,6 +37,59 @@ using Expr = RelayExpr;
 using ExprNode = RelayExprNode;
 using relay::Id;
 
+/*!
+ * \brief Base type of all structure information.
+ *
+ * StructInfo stores possible structure information
+ * deduced during compile-time. It encapsulates
+ * both static type and runtime information such
+ * as shape.
+ *
+ * StructInfo of each non-primitive Expr can be
+ * deduced during compilation in a "best-effort" manner.
+ *
+ * When struct_info appears in function parameter and return
+ * signatures. They will imply a runtime check that matches
+ * the structure information with the value.
+ *
+ * When it appears in normal expressions, they follow "assume-semantics",
+ * which means the compiler will take the deduced information as it is
+ * and do not attempt to do extra proofs.
+ *
+ * Each struct info can be uniquely erased to a static-type.
+ * The compiler will still compile the code(with less information)
+ * when we erase to the static type.
+ *
+ * If an StructInfo contains an Expr field, then that field
+ * must be normalized already through NormalizeArg.
+ * This invariance will be checked in constructors
```

Comment:
“Invariance” means the property of not changing, and “invariant” means the things that do not change. Let’s use “invariant” through out the PR.

(P.S.: Credit to @slyubomirsky’s comment https://github.com/tlc-pack/relax/pull/303#discussion_r1046390692)
```suggestion
 * This invariant will be checked in constructors
```